src/language-server/codegen.ts

```diff
@@ -886,17 +896,32 @@ export function codegenGCVisitTrampoline(
       LLVM._free(loadName);
 
       const args = LLVMUtil.lowerPointerArray([load]);
-      const loweredFieldName = LLVMUtil.lower(`${className}#${name}`);
       LLVM._LLVMBuildCall2(
         llvmBuilder,
         typeRef,
         gcVisitFuncRef,
         args,
         1,
-        loweredFieldName,
+        Number(null) as LLVMStringRef,
       );
-      LLVM._free(loweredFieldName);
       LLVM._free(args);
+
+      // we need to call __whacko_gc_visit
+      if (classType.gcVisitor) {
+        const gcVisitor = classType.gcVisitor as WhackoMethodContext;
+
+        const args = LLVMUtil.lowerPointerArray([
+          LLVM._LLVMGetParam(funcRef, 0),
+        ]);
+        LLVM._LLVMBuildCall2(
+          llvmBuilder,
+          getLLVMFunctionType(LLVM, LLVMUtil, gcVisitor.type),
+          assert(gcVisitor.funcRef, "The function should already be compiled."),
+          args,
+          1,
+          Number(null) as LLVMStringRef,
+        );
+      }
     }
```

Comment:
This should all go outside the loop. You should likely check if this diff is correct...
```suggestion
    }

    // we need to call __whacko_gc_visit
    if (classType.gcVisitor) {
      const gcVisitor = classType.gcVisitor as WhackoMethodContext;

      const args = LLVMUtil.lowerPointerArray([
        LLVM._LLVMGetParam(funcRef, 0),
      ]);
      LLVM._LLVMBuildCall2(
        llvmBuilder,
        getLLVMFunctionType(LLVM, LLVMUtil, gcVisitor.type),
        assert(gcVisitor.funcRef, "The function should already be compiled."),
        args,
        1,
        Number(null) as LLVMStringRef,
      );
    }
```